hcloud/pagination.go

```diff
@@ -0,0 +1,57 @@
+package hcloud
+
+type pageGetter func(start, end int) (*Response, bool, error)
+
+// Page serves as accessor of the API pagination.
+type Page struct {
+	pageGetter  pageGetter
+	err         error
+	currentPage int
+	response    *Response
+}
+
+// Current returns the current page number.
+func (p *Page) Current() int {
+	return p.currentPage
+}
+
+// Response returns the Response of the API request.
+func (p *Page) Response() *Response {
+	return p.response
+}
+
+// GoTo fetches the given page.
+// It returns true on success, or false if the resource is exhausted or an error happened while fetching it.
+func (p *Page) GoTo(page int) bool {
```

Comment:
I find it confusing that `GoTo()` actually fetches the page. I was expecting it to just *set* the next page to fetch (i.e. set the cursor) and `Next()` to actually fetch the page.

Reply:
This would be confusing as well because to get page 6 you would have to call `.GoTo(5)` and then call `.Next()`, so I opted for `GoTo` to fetch the page when run

Reply:
Yes, that would be confusing. `Goto(5).Next()` should fetch page 5. Probably `Next` is not a good name.